src/objmem.cpp

```diff
@@ -90,49 +90,115 @@ void objmemShutdown()
 
 // Check that psVictim is not referred to by any other object in the game. We can dump out some extra data in debug builds that help track down sources of dangling pointer errors.
 #ifdef DEBUG
-#define BADREF(func, line) "Illegal reference to object %d from %s line %d", psVictim->id, func, line
+#define BADREF(func, line) "Illegal reference to object %d from %s line %d in %s[%u]", psVictim->id, func, line, listName, player
 #else
-#define BADREF(func, line) "Illegal reference to object %d", psVictim->id
+#define BADREF(func, line) "Illegal reference to object %d in %s[%u]", psVictim->id, listName, player
 #endif
-static bool checkReferences(BASE_OBJECT *psVictim)
+
+static bool _checkStructReferences(BASE_OBJECT *psVictim, const PlayerObjectList<STRUCTURE>& psPlayerStructList, unsigned player, const char* listName)
```

Comment:
You can use the existing `StructureList` typedef here.

Reply:
Ah, great catch - missed those on first glance. 👍